### PR TITLE
[CORE-1130] Reorg Pipelines docs

### PR DESCRIPTION
**Pull request by infraredgirl**

### CODE REVIEW

**Review by ebeneliason — COMMENTED** [review]
This looks good! I made some suggested edits to the callouts.

It's a little odd that the whole sidebar is essentially the same across versions, which means that when you click the link in the new sidebar to go to the legacy version, the fact that you've entered a whole new area isn't obvious. That meets the AC as specified, though, and I'm not sure there's a better approach. Also, over time they may diverge which will make it more obvious.

I'm hesitant regarding the `pipelines-v1` URL path. I worry about conflating the actual release version with this higher level abstraction. Also, I think conceptually we want it to live _inside_ the top level pipelines section. That might suggest a path such as `_docs-sources/pipelines/legacy/how-it-works/index.md` and so on.

(We're not using "legacy" in the docs themselves, but perhaps it's reasonable to use that in the URL slug? Alternatively we could use "ecs-deploy-runner" but that could then be conflated with the module docs itself.)

@josh-padnick FYI, let us know if you have opinions on the organization/naming here.

File: _docs-sources/pipelines-v1/overview/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# ECS Deploy Runner
+
+:::info New Version
+This is documentation for an old version of Gruntwork Pipelines. Looking for the documentation for the most recent version? Click [here](../../pipelines/overview/).
+:::
```
[the remark this answers is not in the document]
Comment (infraredgirl) [reply _docs-sources/pipelines-v1/overview/index.md]: Fixed in 288fd46.

File: _docs-sources/pipelines/overview/index.md [hunk 2]
```diff
@@ -1,16 +1,7 @@
...
-This makes it trivial for anyone with access to your CI/CD system to access AWS credentials with permissions
-greater than they might otherwise need.
-Gruntwork Pipelines allows a highly restricted set of permissions to be supplied to the CI/CD tool while
-infrastructure related permissions reside safely within your own AWS account. This reduces the exposure of your
-high value AWS secrets.
-
-
+:::info Previous Version
+Looking for the documentation for the previous version of Gruntwork Pipelines? Click [here](../../pipelines-v1/overview/).
+:::
```
[the remark this answers is not in the document]
Comment (infraredgirl) [reply _docs-sources/pipelines/overview/index.md]: Fixed in 288fd46.

**Comment by josh-padnick** [comment]
Agreed, this looks great, overall! +1 on Eben's points. Here are some of my thoughts:

1. This should say "What is ECS Deploy Runner?" In general, we want to refer to the legacy product as "ECS Deploy Runner" and the newer product as "Gruntwork Pipelines" or just "Pipelines."
   CleanShot 2023-08-11 at 15 27 14@2x

2. On the new Pipelines table of contents, I like the "Getting Started" header, but maybe the doc there should be less about "single account tutorial" and more "Try out Pipelines," where the focus is on helping the user stand it up and try out the UX?

Beyond that, I'm sure we'll populate content as we build it out. It's very exciting to see this!

**Comment by infraredgirl** [comment]
> 1. This should say "What is ECS Deploy Runner?"

Done

> In general, we want to refer to the legacy product as "ECS Deploy Runner" and the newer product as "Gruntwork Pipelines" or just "Pipelines."
 
There is a multitude of references to "pipelines" everywhere in the old docs. I've replaced references to pipelines to ECS Deploy Runner in the sidebar / headers, but not in the rest of the old docs. Happy to go through the entire docs and fix it everywhere, but I was not sure if we want a deeper edit to the old docs as part of this story - opinions welcome.

> 2. On the new Pipelines table of contents, I like the "Getting Started" header, but maybe the doc there should be less about "single account tutorial" and more "Try out Pipelines," where the focus is on helping the user stand it up and try out the UX?

The content of the new pipelines docs is out of scope for this story, so I'll leave those changes for the future.
 
> **Update:** For the legacy pipelines URL, I'd go with `/ecs-deploy-runner` as well, per the first point.

Done

**Comment by MoonMoon1919** [comment]
We merged this into the `devops-foundations` branch this morning. Closing in favor of that PR.

**Pull request closed without merging**